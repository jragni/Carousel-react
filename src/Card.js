import React from "react";
import "./Card.css";

/**Card
 *
 * Props:
 * - src
 * - caption
 * - currNum
 * - totalNum
 *
 * State:
 *    None
 *
 * Card -> Carousel
 */
function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}

export default Card;
