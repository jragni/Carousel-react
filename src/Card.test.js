/** Tests for Card Component */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", () => {
  render(<Card />);
});

it("has correct src, caption, currNum, and totalNum", () => {
  const { debug, container } = render(
    <Card caption="testCaption" src="testSrc" currNum="2" totalNum="5" />
  );
  // query tags
  const smallHTML = container.querySelector(".Card-small");
  const cardTitleHTML = container.querySelector(".Card-title");
  const imgHTML = container.querySelector(".Card-image");
  const caption = imgHTML.getAttribute("alt");
  const src = imgHTML.getAttribute("src");
  const smallText = smallHTML.textContent;
  const cardTitle = cardTitleHTML.textContent;

  expect(caption).toEqual("testCaption");
  expect(src).toEqual("testSrc");
  expect(smallText).toContain("2");
  expect(cardTitle).toEqual("testCaption");
  expect(smallText).toContain("5");
});

it("mathces snapshots", () => {
  const { container, debug } = render(<Card />);
  expect(container).toMatchSnapshot();
});

//TODO: test interactions with Carousel
