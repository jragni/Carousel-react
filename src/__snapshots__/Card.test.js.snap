// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathces snapshots 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image 
       of 
      .
    </small>
  </div>
</div>
`;
